Chisel/CHLFunctions.h

```diff
@@ -0,0 +1,15 @@
+//
+//  CHLFunctions.h
+//  Chisel
+//
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import <malloc/malloc.h>
+
+/**
+ * Enumerates all live memory allocations by calling `block` for each malloc range (represented as a `vm_range_t`).
+ */
+void CHLEnumerateAllocationsWithBlock(void (^block)(vm_range_t range));
```

Comment:
If you mean nullability identifiers, you could:

1. Wrap the header in `NS_ASSUME_NONNULL`.
2. Add `nullable` or `_Nullable` (or `__nullable` if you're supporting Xcode 6.4 and prior) to all nullable parameters and return values.

You should still check for null in the implementation. Without certain warnings enabled, it's very easy for users to accidentally pass a nullable pointer to a parameter marked as nonnull; see: http://modocache.io/clang-warn-nullable-to-nonnull-conversion. Even with those warnings enabled, it's still pretty easy, although newer versions of the Clang static analyzers may catch more of these cases.